lib/puppet/util.rb

```diff
@@ -349,6 +356,52 @@ def uri_to_path(uri)
   end
   module_function :uri_to_path
 
+  private
+
+  RFC_3986_URI_REGEX = /^(?<scheme>([^:\/?#]+):)?(?<authority>\/\/([^\/?#]*))?(?<path>[^?#]*)(\?(?<query>[^#]*))?(#(?<fragment>.*))?$/
+
+  public
+
+  # Will escape a query parameter or any portion of the query string
+  # so that it will remain unescapable with URI.unescape
```

Comment:
Minor but any chance you wouldn't mind rewording this comment to something like `so that it can still be unescaped with URI.unescape`? I originally read this as saying the opposite of what's intended, ie "so that cannot be escaped with URI.unescape"